tests/codegen/vec-calloc.rs

```diff
@@ -42,6 +42,43 @@ pub fn vec_one_bytes(n: usize) -> Vec<u8> {
     vec![1; n]
 }
 
+// CHECK-LABEL: @vec_false
+#[no_mangle]
+pub fn vec_false(n: usize) -> Vec<bool> {
+    // CHECK-NOT: call {{.*}}alloc::vec::from_elem
+    // CHECK-NOT: call {{.*}}reserve
+    // CHECK-NOT: call {{.*}}__rust_alloc(
+    // CHECK-NOT: call {{.*}}llvm.memset
+
+    // CHECK: call {{.*}}__rust_alloc_zeroed(
+
+    // CHECK-NOT: call {{.*}}alloc::vec::from_elem
+    // CHECK-NOT: call {{.*}}reserve
+    // CHECK-NOT: call {{.*}}__rust_alloc(
+    // CHECK-NOT: call {{.*}}llvm.memset
+
+    // CHECK: ret void
+    vec![false; n]
+}
+
+// CHECK-LABEL: @vec_true
+#[no_mangle]
+pub fn vec_true(n: usize) -> Vec<bool> {
+    // CHECK-NOT: call {{.*}}alloc::vec::from_elem
+    // CHECK-NOT: call {{.*}}reserve
+    // CHECK-NOT: call {{.*}}__rust_alloc_zeroed(
+
+    // CHECK: call {{.*}}__rust_alloc(
+    // CHECK: call {{.*}}llvm.memset
```

Comment:
So, interestingly, this test looks like it passes without this PR: <https://rust.godbolt.org/z/G6es4qqsa>

It has `__rust_alloc_zeroed` for `vec_false`, and the `llvm.memset` for `vec_true`.

Is there anything else it could check that would make it catch someone removing the extra specialization and saying "well, all the codegen tests still pass"?